### PR TITLE
[BTS-274] button

### PR DESCRIPTION
**Pull request by ggdaltoso**

### CODE REVIEW

File: components/Button/Button.jsx [hunk 1]
```diff
@@ -1,91 +1,329 @@
...
+
+  ${({ size, theme: { baseFontSize } }) => {
+    const sizes = {
+      xsmall: `${baseFontSize * 0.75}px`,
+      small: `${baseFontSize * 0.75}px`,
+      medium: `${baseFontSize}px`,
+      large: `${baseFontSize * 1.25}px`,
+      xlarge: `${baseFontSize * 1.5}px`,
+    };
+    return `font-size: ${sizes[size] || sizes.medium};`;
```
Comment (alan-oliv) [inline components/Button/Button.jsx]: `|| sizes.medium` - Why is this needed (since defaultProps has size: 'medium') ?
(Same doubt in `height` and `padding` props)

File: components/Button/Button.jsx [hunk 2]
```diff
@@ -1,91 +1,329 @@
...
+  }}
+
+  ${({
+    size,
+    theme: {
+      spacing: { large, xlarge, xxlarge, xxxlarge },
+    },
+  }) => {
+    const heights = {
```
Comment (alan-oliv) [inline components/Button/Button.jsx]: Is this property really needed? Can we solve with padding only?
If can't, IMHO I'd avoid using spacing values for component height

File: components/Button/Button.jsx [hunk 3]
```diff
@@ -1,91 +1,329 @@
...
+      background-color: ${bgColor};
+      color: ${textColor};
+
+      border: 1.5px solid ${disabled ? neutral500 : mainColor500};
+
+      ${shadow(2, neutral500)({ theme })}
+
+      :hover {
+        ${
+          !disabled
```
Comment (alan-oliv) [inline components/Button/Button.jsx]: What do you guys think of using an logical and (&&) here instead of a ternary, since has no else condition?
(in `focus` and `active` too)

File: components/Button/Button.jsx [hunk 4]
```diff
@@ -1,91 +1,329 @@
...
+      ${({
+        size,
+        theme: {
+          spacing: { medium, large },
+        },
+      }) => {
+        const sizes = {
+          xsmall: `${medium}px`,
+          small: `${medium}px`,
+          normal: `${large}px`,
```
Comment (alan-oliv) [inline components/Button/Button.jsx]: I guess we can update this object and rename `normal` to `medium` to keep with our nomenclature convention

File: components/Button/Button.jsx [hunk 5]
```diff
@@ -1,91 +1,329 @@
...
+  ${props =>
+    props.center &&
+    `
+    margin-left: auto;
+    margin-right: auto;
+  `}
 
   ${ButtonIcon} {
-    ${iconSize}
+    margin-right: 5px;
```
Comment (ddsilva) [inline components/Button/Button.jsx]: Is `spacing.xxsmall` (4px) not electable in place of this hard coded value?

File: components/Button/Button.jsx (hunk 1 above)
Comment (ggdaltoso) [reply components/Button/Button.jsx]: There is no need for this. We get the point. :+1:

File: components/Button/Button.jsx (hunk 2 above)
Comment (ggdaltoso) [reply components/Button/Button.jsx]: _Is this property really needed?_
Yes. It is needed.

_Can we solve with padding only?_
Yes, we can.

_If can't, IMHO I'd avoid using spacing values for component height_
We follow material design (MD) guidelines as a reference for this. In this case, the MD uses height to define the vertical size of the component.

File: components/Button/Button.jsx (hunk 5 above)
Comment (ggdaltoso) [reply components/Button/Button.jsx]: Yes, it is :clap: 

@allyssonsantos

File: components/Button/Button.jsx (hunk 3 above)
Comment (ggdaltoso) [reply components/Button/Button.jsx]: Using `&&` will return a boolean condition what will make a string template to be like this:
 ```
   :hover { false; }
```
We use a ternary expression to prevent that situation.

File: components/Button/Button.jsx (hunk 4 above)
Comment (ggdaltoso) [reply components/Button/Button.jsx]: :+1:

File: components/Button/Button.jsx (hunk 2 above)
Comment (alan-oliv) [reply components/Button/Button.jsx]: If we can solve with padding, so it is not needed (respectin')
I'd remove this part, but i'll leave to you guys to decide or vote <3

Comment (ddsilva) [reply components/Button/Button.jsx]: Although Material Design specifies that this is solved with height, it's a common sense among us (including me) that padding it's better for this case. So, is there any problem to implements in this way?
